src/ILLink.RoslynAnalyzer/RequiresUnreferencedCodeAnalyzer.cs

```diff
@@ -32,8 +34,28 @@ public sealed class RequiresUnreferencedCodeAnalyzer : RequiresAnalyzerBase
 				operationContext.Operation.Syntax.GetLocation ()));
 		};
 
+		private Action<SymbolAnalysisContext> typeDerivesFromRucBase {
+			get {
+				return symbolAnalysisContext => {
+					if (symbolAnalysisContext.Symbol is INamedTypeSymbol typeSymbol && !typeSymbol.HasAttribute (RequiresUnreferencedCodeAttribute)) {
+						if (typeSymbol.BaseType is INamedTypeSymbol baseType && baseType.HasAttribute (RequiresUnreferencedCodeAttribute)) {
+							if (baseType.TryGetAttribute (RequiresUnreferencedCodeAttribute, out var requiresUnreferencedCodeAttribute)) {
```

Comment:
Also not sure if we need the check 'is INamedTypeSymbol' for the symbolAnalysisContext.Symbol since you register the SymbolAnalysisContext per SymbolKind.NamedType it should always be a INamedTypeSymbol. For BaseType is fine since is doing the null check

Reply:
It looks like we'd still need to cast the Symbol to an `INamedTypeSymbol` to access the `BaseType` property. I'm not exactly sure, but I think a cast would do the same runtime checks as the `symbolAnalysisContext.Symbol is INamedTypeSymbol typeSymbol` and I personally prefer that notation. Let me know if I'm wrong about that, though.

Reply:
You are right in the Analyzer base we cast the symbol disregard my comment about that